lightning/src/routing/network_graph.rs

```diff
@@ -1765,4 +2154,647 @@ mod tests {
 		network.write(&mut w).unwrap();
 		assert!(<NetworkGraph>::read(&mut ::std::io::Cursor::new(&w.0)).unwrap() == *network);
 	}
+
+	#[test]
+	fn sending_query_channel_range() {
+		let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+		let node_privkey_1 = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_privkey_2 = &SecretKey::from_slice(&[41; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_privkey_1);
+		let node_id_2 = PublicKey::from_secret_key(&secp_ctx, node_privkey_2);
+
+		let chain_hash = genesis_block(Network::Testnet).header.block_hash();
+		let first_blocknum = 0;
+		let number_of_blocks = 0xffff_ffff;
+
+		// When no active query exists for the node, it should send a query message and generate a task
```

Comment:
I wonder if the active query be time-out at some point in case of irresponsive peer ?

Reply:
Definitely agree on this point.  Also need to clean up tasks on peer disconnection.